pyside 5.12.0

albertosottile opened the pull request:
Created with `brew bump-formula-pr`.

albertosottile commented:
Fails on all the platforms. The argument `setup.py` is missing from the invocation made by `Language::Python.setup_install_args`, which is:
`/usr/local/opt/python/bin/python3.7 -c --no-user-cfg install --prefix=/usr/local/Cellar/pyside/5.12.0 --install-scripts=/usr/local/Cellar/pyside/5.12.0/bin --single-version-externally-managed --record=installed.txt --install-lib /usr/local/Cellar/pyside/5.12.0/lib/python3.7/site-packages --ignore-git --no-examples --macos-use-libc++ --jobs=3 --install-scripts /usr/local/Cellar/pyside/5.12.0/bin --internal-build-type=shiboken2`

`setup.py` should be after `-c`. I am not sure about how to fix this, since running `setup.py` manually works, even while in superenv.

SMillerDev commented:
Could you report that issue to brew?

MikeMcQuaid commented:
> `setup.py` should be after `-c`. I am not sure about how to fix this, since running `setup.py` manually works, even while in superenv.

Have you tried manually adding it to the invoked command?

albertosottile commented:
> Have you tried manually adding it to the invoked command?

Once the build fails in `--debug` mode, I tried to run the command from the console (while in `superenv`) and the build works. I could get rid of `Language::Python.setup_install_args` in the formula and make it run the command manually, if that's the thing you are suggesting.

MikeMcQuaid commented:
I'm suggesting you add the `setup.py` to the `args` or similar if it's now required and wasn't previously.

albertosottile commented:
It was required also in previous versions, but `Language::Python.setup_install_args` took care of it (and also of some other arguments). I will try to add it explicitly.

albertosottile commented:
I changed the formula to add "setup.py" explicitly in the `python3` invocation, but I still get a BuildError similar to before
```
BuildError: Failed executing: python3 setup.py -c import\ setuptools,\ tokenize
__file__\ =\ 'setup.py'
exec(compile(getattr(tokenize,\ 'open',\ open)(__file__).read()
\ \ .replace('\r\n',\ '\n'),\ __file__,\ 'exec'))
 --no-user-cfg install --prefix=/usr/local/Cellar/pyside/5.12.0 --install-scripts=/usr/local/Cellar/pyside/5.12.0/bin --single-version-externally-managed --record=installed.txt --install-lib /usr/local/Cellar/pyside/5.12.0/lib/python3.7/site-packages --ignore-git --no-examples --macos-use-libc++ --jobs=4 --install-scripts /usr/local/Cellar/pyside/5.12.0/bin
```

Am I allowed to remove `Language::Python.setup_install_args` from the formula and add the extra arguments it introduces manually? (` --no-user-cfg install --prefix=/usr/local/Cellar/pyside/5.12.0 --install-scripts=/usr/local/Cellar/pyside/5.12.0/bin --single-version-externally-managed --record=installed.txt`)

MikeMcQuaid commented:
> Am I allowed to remove `Language::Python.setup_install_args` from the formula and add the extra arguments it introduces manually?

I'd rather that be a last resort and we try to figure out what's changed such that it's no longer working.

albertosottile commented:
> I'd rather that be a last resort and we try to figure out what's changed such that it's no longer working.

Is there anything I can do to help you?

MikeMcQuaid commented:
Sorry, to be clear: I'm not investigating this currently. Yes, if you could figure out what changed to break this that would be good.

tdsmith commented:
It looks like setup.py got rewritten between 5.11.2 (from something that just calls `setup()`) and 5.12.0 (which does elaborate magic), which is why there's a new failure mode.

`--no-examples` is not mentioned as a legal flag in the 5.12.0 version and neither is `--jobs`; does dropping those help or is it choking on the flags we want setuptools to interpret?

tdsmith commented:
@albertosottile points out that setup.py runs fine with those flags when it's invoked directly, so the problem is probably that the elaborate magic in the current version of pyside's setup.py is incompatible with how we're invoking it.

We're doing the same thing pip does to invoke setup.py's -- pyside2 only ships wheels (precompiled binary distributions that don't have a setup.py) on pypi and not sdists, which is a signal that pyside2's setup.py may not be compatible with pip, either.

Given that we have a file named "setup.py" that doesn't really behave like a conventional setup.py, it probably makes sense not to use the usual wrapper here.

MikeMcQuaid commented:
> Given that we have a file named "setup.py" that doesn't really behave like a conventional setup.py, it probably makes sense not to use the usual wrapper here.

👍 thanks again @tdsmith.

@albertosottile You can the use of the wrapper in favour of hardcoding it.

albertosottile commented:
I removed the `Language::Python.setup_install_args` wrapper and added the arguments it introduced in the `setup.py` invocation. Now the PR passes all the tests. Thanks @tdsmith for your help.

MikeMcQuaid commented on Formula/pyside.rb:
```diff
@@ -17,27 +17,39 @@ class Pyside < Formula
+      --record=installed.txt
+    ]
+
     args = %W[
```
Can you combine these two arrays into a single array including `--install-lib` and `site-packages`?

albertosottile replied:
The destination path is different for `python` and `python@2`. Would you prefer that I define two args variables or reuse the same array name and change the `--install-lib` argument?

albertosottile replied:
Done, thanks for reviewing this.

albertosottile commented:
No idea of what happened on High Sierra, especially because I make all my local tests on 10.13.6. Is there a way to run again the check?

MikeMcQuaid commented:
Rerunning now

MikeMcQuaid commented:
Thanks so much for your contribution and all the back and forth! Without people like you submitting PRs we couldn't run this project. You rock, @albertosottile!

albertosottile commented:
No, thank you for running this amazing project!

alcroito commented:
Once https://codereview.qt-project.org/#/c/252315/ goes in, it should be possible to use the standard "Language::Python.setup_install_arg" approach. That would be released with version 5.12.2.

MikeMcQuaid commented:
@alcroito Good news, thanks.
